new option --check-source

auerswal opened the pull request:
Using the new option --check-source discards Echo reply packets sourced from an address that is different from the specified target address.  This can be useful on busy monitoring hosts, because it is possible that two fping processes running at he same time use the same ICMP Echo Reply Identifier for ICMP Echo messages sent to different hosts.

This aims to address GH issue #206.

auerswal committed:
new option --check-source

Using the new option --check-source discards Echo reply packets
sourced from an address that is different from the specified target
address.  This can be useful on busy monitoring hosts, because it
is possible that two fping processes running at he same time use
the same ICMP Echo Reply Identifier for ICMP Echo messages sent to
different hosts.

This aims to address GH issue #206.

gsnw reviewed (approved):
I see no problem in adopting this

auerswal committed:
Catch long-only option error when using -0

The long-only options use the character '0' to select the switch
case used to handle the different long-only options.  This can be
given as a normal short option that requires an argument:

    $ fping -0
    fping: option requires an argument -- '0'
    see 'fping -h' for usage information
    $ fping -0 127.0.0.1 127.0.0.2
    127.0.0.2 is alive

Catch this usage error by requiring that one of the long-only
options is given when handling the case of '0'.

auerswal commented:
I do not understand why the test fails on the GitHub CI, it works on my Ubuntu 20.04 system. I am closing this pull request.
